api/hypershift/v1beta1/azure_types.go

```diff
@@ -1,144 +1,177 @@
 package v1beta1
 
+// AzureVMImageType is used to specify the source of the Azure VM boot image.
+// Valid values are ImageID and AzureMarketplace.
+// +kubebuilder:validation:Enum:=ImageID;AzureMarketplace
 type AzureVMImageType string
 
 const (
-	ImageID          AzureVMImageType = "ImageID"
+	// ImageID is the used to specify that an Azure resource ID of a VHD image is used to boot the Azure VMs from.
+	ImageID AzureVMImageType = "ImageID"
+
+	// AzureMarketplace is used to specify the Azure Marketplace image info to use to boot the Azure VMs from.
 	AzureMarketplace AzureVMImageType = "AzureMarketplace"
 )
 
+// AzureNodePoolPlatform is the platform specific configuration for an Azure node pool.
 type AzureNodePoolPlatform struct {
-	// VMSize is the Azure VM instance type to use for the nodes being created in the nodepool.
+	// vmSize is the Azure VM instance type to use for the nodes being created in the nodepool.
 	//
+	// TODO: This should be validated with a regex, based on https://learn.microsoft.com/en-us/azure/virtual-machines/vm-naming-conventions
 	// +kubebuilder:validation:Required
-	VMSize string `json:"vmsize"`
+	VMSize string `json:"vmSize"`
 
-	// ImageID is the id of the image to boot from. If unset, the default image at the location below will be used and
+	// image is used to configure the VM boot image. If unset, the default image at the location below will be used and
 	// is expected to exist: subscription/<subscriptionID>/resourceGroups/<resourceGroupName>/providers/Microsoft.Compute/images/rhcos.x86_64.vhd.
 	// The <subscriptionID> and the <resourceGroupName> are expected to be the same resource group documented in the
 	// Hosted Cluster specification respectively, HostedCluster.Spec.Platform.Azure.SubscriptionID and
 	// HostedCluster.Spec.Platform.Azure.ResourceGroupName.
+	// The image is immutable once set.
```

Comment:
Explain all validations to users

Reply:
I believe I set it to immutable because I thought changing it would result in a rollout of new nodes, but maybe that is not true or maybe its okay to allow that if the user choose to do so 🤔

Reply:
That's fine, whatever choice is made we need to tell them why it is immutable. If we can tell them the reason it's immutable that's even better

Reply:
In this case, actually we may want this to be mutable, but warn users that if they do change it, it will roll out, WDYT?

Reply:
sgtm

Reply:
This needs to be mutable and trigger a rollout.
Let's use // +noderollout for fields which do so. That should be in the upper level NodePool struct (.NodePoolPlatform ) though since any change to AzureNodePoolPlatform triggers a rollout by design and if anything within it is immutable it needs to be enforced by CEL and documented here

Reply:
Is there any tooling looking for the `+noderollout` marker or is that just to jog memory?